framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1173,13 +1174,17 @@ public void addAnnotation(AnnotationMirror a) {
         /**
          * Sets the parameter types of this executable type, excluding the receiver.
          *
-         * @param params the parameter types, excluding the receiver
+         * @param params a list of parameter types to be captured by this method, excluding the
+         *     receiver
          */
-        /*package-private*/ void setParameterTypes(List<? extends AnnotatedTypeMirror> params) {
-            paramTypes =
-                    params.isEmpty()
-                            ? Collections.emptyList()
-                            : Collections.unmodifiableList(new ArrayList<>(params));
+        /*package-private*/ void setParameterTypes(List<AnnotatedTypeMirror> params) {
+            if (params.isEmpty()) {
+                paramTypes = Collections.emptyList();
+            } else if (CollectionUtils.isUnmodifiableCollection(params)) {
```

Comment:
I'm wondering whether we really need this check. There are very few calls of these methods - couldn't we just change them to already pass an unmodifiable collection and not add this extra test?

Reply:
Should the caller decide whether `paramTypes` is a modifiable list or not? I personally want to treat it as an implied rule in `AnnotatedTypeMirror`.

Reply:
Yes, the rule should be that an unmodifiable collection must be passed in.
But we control all call sites and should just ensure that, instead of checking whether it's the case and wrapping the collection.

Reply:
Ok, I've updated the call sites.